### PR TITLE
feat: Safe kit testing

### PR DESCRIPTION
**Pull request by yagopv**
## What it solves
Resolves https://github.com/safe-global/safe-core-sdk/issues/869

## How this PR fixes it
This PR add unit testing to `safe-kit`

### CODE REVIEW

File: packages/safe-kit/src/SafeClient.ts [hunk 1]
```diff
@@ -38,15 +41,25 @@ export class SafeClient {
...
   /**
    * Sends transactions through the Safe protocol.
+   * You can send an array to transactions { to, value, data} that we will convert to a transaction batch
    *
-   * @param {TransactionBase[]} transactions An array of transactions to be sent.
-   * @param {TransactionOptions} [options] Optional transaction options.
+   * @param {SendTransactionProps} props The SendTransactionProps object.
+   * @param {TransactionBase[]} props.transactions An array of transactions to be sent.
+   * @param {string} props.transactions.to The recipient address of the transaction.
+   * @param {string} props.transactions[].value The value of the transaction.
```
Comment (dasanra) [inline packages/safe-kit/src/SafeClient.ts:49]:
```suggestion
   * @param {string} props.transactions[].to The recipient address of the transaction.
   * @param {string} props.transactions[].value The value of the transaction.
```

File: packages/safe-kit/src/types.ts [hunk 2]
```diff
@@ -1,6 +1,50 @@
...
+  validUntil?: number
+  validAfter?: number
+  feeEstimator?: IFeeEstimator
+}
+
+export type ConfirmSafeOperationProps = {
+  safeOperationHash: string
+}
+
+export type PaginationOptions = {
```
Comment (dasanra) [inline packages/safe-kit/src/types.ts:43]: Should this be the same type used in the api-kit? As it's going to be used to call the api-kit

File: packages/safe-kit/src/extensions/messages/SafeMessageClient.ts [hunk 3]
```diff
@@ -77,7 +83,7 @@ export class SafeMessageClient {
    * @param {GetSafeMessageListProps} [options] Optional query parameters for pagination
    * @returns {Promise<SafeMessageListResponse>} A list of pending messages
    */
-  async getPendingMessages(options?: GetSafeMessageListProps): Promise<SafeMessageListResponse> {
+  async getPendingMessages(options?: PaginationOptions): Promise<SafeMessageListResponse> {
```
Comment (dasanra) [inline packages/safe-kit/src/extensions/messages/SafeMessageClient.ts:86]: Should this function also get an object with optional parameters?

Comment (yagopv) [reply packages/safe-kit/src/extensions/messages/SafeMessageClient.ts:86]: apart from the pagination options ? I can only see a parameter to get a concrete message and in this case can be a explicit method. What parameters are you thinking in?

File: packages/safe-kit/src/types.ts (hunk 2 above)
Comment (yagopv) [reply packages/safe-kit/src/types.ts]: yes, now that i removed the `order` it doesn't make sense the explicit type

Comment (yagopv) [reply packages/safe-kit/src/types.ts]: done

**Review by dasanra — APPROVED** [review]

**Pull request merged**